File: HRMS.Desktop/lib/models/employee_position.dart
import 'package:json_annotation/json_annotation.dart';

import 'employee.dart';
import 'position.dart';

part 'employee_position.g.dart';

@JsonSerializable()
class EmployeePosition {
  int id;
  Employee? employee;
  Position? position;
  DateTime startDate;
  DateTime? endDate;
  double salary;
  int vacationDays;
  int employmentType;
  String workingHours;

  EmployeePosition(
    this.id,
    this.employee,
    this.position,
    this.startDate,
    this.endDate,
    this.salary,
    this.vacationDays,
    this.employmentType,
    this.workingHours,
  );

  factory EmployeePosition.fromJson(Map<String, dynamic> json) =>
      _$EmployeePositionFromJson(json);

  Map<String, dynamic> toJson() => _$EmployeePositionToJson(this);
}

File: HRMS.Core/Models/EmployeePosition.cs
﻿using HRMS.Core.Models.Enums;

namespace HRMS.Core.Models;

public class EmployeePosition : Base
{
    public Employee Employee { get; set; } = new();

    public Position Position { get; set; } = new();

    public DateTime StartDate { get; set; }

    public DateTime? EndDate { get; set; }

    public decimal Salary { get; set; }

    public int VacationDays { get; set; }

    public EmploymentType EmploymentType { get; set; }

    public string WorkingHours { get; set; } = string.Empty;
}